sdk/storage/inc/datalake/response_models.hpp

```diff
@@ -0,0 +1,168 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include "blobs/internal/protocol/blob_rest_client.hpp"
+#include "protocol/datalake_rest_client.hpp"
+
+namespace Azure { namespace Storage { namespace Files { namespace DataLake {
+
+  // ServiceClient models:
+
+  using ListFileSystemsResult = ServiceListFileSystemsResponse;
+
+  // FileSystemClient models:
+
+  struct FileSystemProperties
+  {
+    std::string ETag;
+    std::string LastModified;
+    std::map<std::string, std::string> Metadata;
+  };
+  using FileSystemInfo = FileSystemCreateResponse;
+
+  // PathClient models:
+
+  struct Acl
+  {
+    std::string Scope;
+    std::string Type;
+    std::string Id;
+    std::string Permissions;
+
+    /**
+     * @brief Creates an Acl based on acl input string.
+     * @param aclString the string to be parsed to Acl.
+     * @return Acl
+     */
+    static Acl FromString(const std::string& aclString);
+
+    /**
+     * @brief Creates a string from an Acl.
+     * @param acl the acl object to be serialized to a string.
+     * @return std::string
+     */
+    static std::string ToString(const Acl& acl);
+
+    /**
+     * @brief Creates a vector of Acl from a string that indicates multiple acls.
+     * @param dataLakeAclsString the string that contains multiple acls.
+     * @return std::vector<Acl>
+     */
+    static std::vector<Acl> DeserializeAcls(const std::string& dataLakeAclsString);
+
+    /**
+     * @brief Creates a string that contains several Acls.
+     * @param dataLakeAclsArray the acls to be serialized into a string.
+     * @return std::string
+     */
+    static std::string SerializeAcls(const std::vector<Acl>& dataLakeAclsArray);
+  };
+
+  struct PathProperties
+  {
+    std::string ETag;
+    std::string LastModified;
+    std::string CreationTime;
+    std::map<std::string, std::string> Metadata;
+    Azure::Core::Nullable<std::string> LeaseDuration;
+    Azure::Core::Nullable<LeaseStateType> LeaseState;
+    Azure::Core::Nullable<LeaseStatusType> LeaseStatus;
+    DataLakeHttpHeaders HttpHeaders;
+    Azure::Core::Nullable<bool> ServerEncrypted;
+    Azure::Core::Nullable<std::string> EncryptionKeySHA256;
+    Azure::Core::Nullable<bool> AccessTierInferred;
+    Azure::Core::Nullable<std::string> AccessTierChangeTime;
+    Azure::Core::Nullable<std::string> CopyId;
+    Azure::Core::Nullable<std::string> CopySource;
+    Azure::Core::Nullable<Blobs::CopyStatus> CopyStatus;
+    Azure::Core::Nullable<std::string> CopyProgress;
+    Azure::Core::Nullable<std::string> CopyCompletionTime;
+  };
+
+  struct PathAccessControl
+  {
+    std::string ETag;
+    std::string LastModified;
+    std::vector<Acl> Acls;
+  };
+
+  struct SetPathHttpHeadersResponse
+  {
+    std::string ETag;
+    std::string LastModified;
+    Azure::Core::Nullable<int64_t> SequenceNumber;
```

Comment:
What is this `SequenceNumber`?

Reply:
Returned by blob service: https://docs.microsoft.com/en-us/rest/api/storageservices/set-blob-properties

Reply:
Is datalake able to interop with page blob?

Reply:
Removed.